core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -262,7 +262,13 @@ object SparkEnv extends Logging {
         rpcEnv.asInstanceOf[AkkaRpcEnv].actorSystem
       } else {
         // Create a ActorSystem for legacy codes
-        AkkaUtils.createActorSystem(actorSystemName, hostname, port, conf, securityManager)._1
+        AkkaUtils.createActorSystem(
+          actorSystemName + "ActorSystem",
+          hostname,
+          rpcEnv.address.port + 1, // Use a new port to avoid port conflicts
```

Comment:
This is not necessary for fixing this issue. Just to avoid trying to bind on a used port. I tested locally without this change and it still worked.